torch/utils/mkldnn.py

```diff
@@ -1,43 +1,160 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
-import functools
 
 import torch
 
 
-def to_mkldnn(module):
-    def t_fn(t):
-        if t.is_floating_point():
-            return t.to_mkldnn()
+class MkldnnLinear(torch.jit.ScriptModule):
+    def __init__(self, dense_module):
+        super(MkldnnLinear, self).__init__()
+        self.register_buffer('weight', dense_module.weight.to_mkldnn())
+        if dense_module.bias is not None:
+            self.has_bias = torch.jit.Attribute(True, bool)
+            self.register_buffer('bias', dense_module.bias.to_mkldnn())
+        else:
+            self.has_bias = torch.jit.Attribute(False, bool)
+            # TODO: Remove this once ScriptModule supports registering None buffer
+            self.register_buffer(
+                'bias',
+                torch.zeros([dense_module.weight.size(0)], dtype=torch.float).to_mkldnn())
+
+    @torch.jit.script_method
+    def __getstate__(self):
+        return (self.weight.to_dense(), self.bias.to_dense(), self.has_bias)
```

Comment:
I wish there was a workaround that require adding additional field. Let's land it for now but please create a lockdown issue with example code you want to have  working, ok?